tests/codegen/union-abi.rs

```diff
@@ -54,7 +54,7 @@ pub fn test_UnionF32F32(_: UnionF32F32) -> UnionF32F32 { loop {} }
 
 pub union UnionF32U32{a:f32, b:u32}
 
-// CHECK: define {{(dso_local )?}}i32 @test_UnionF32U32(i32{{( %0)?}})
+// CHECK: define noundef {{(dso_local )?}}i32 @test_UnionF32U32(i32 noundef{{( %0)?}})
```

Comment:
I don't know if this is legal...

Reply:
Added a check so that `noundef` is only emitted for array immediates, not anything else.